pkg/secrets/config/config.go

```diff
@@ -49,13 +54,27 @@ func NewFromEnv() (*Config, error) {
 		return nil, err
 	}
 
+	retryInterval := ParseIntFromEnvOrDefault("RETRY_INTERVAL", DEFAULT_RETRY_INTERVAL)
+
+	retryCountLimit := ParseIntFromEnvOrDefault("RETRY_COUNT_LIMIT", DEFAULT_RETRY_COUNT_LIMIT)
+
 	return &Config{
 		PodNamespace:       podNamespace,
 		RequiredK8sSecrets: requiredK8sSecrets,
+		RetryCountLimit:    retryCountLimit,
+		RetryInterval:      retryInterval,
 		StoreType:          storeType,
 	}, nil
 }
 
+func ParseIntFromEnvOrDefault(environmentVariable string, defaultVariable int) int {
+	retry := defaultVariable
+	if envVar, _ := strconv.Atoi(os.Getenv(environmentVariable)); envVar != defaultVariable && envVar != 0 {
```

Comment:
instead of `envVar != 0`, which is a valid value, use `err != nil`, which indicated parsing error occurred.
Also, I suggest to document that the we failed to parse the value, what do you think?